### PR TITLE
Istio 1.19.0

### PR DESCRIPTION
**Pull request by nicholasSUSE**
Istio update append 1.19.0
Kiali update append 1.72
Jaeger update append 1.49

Related PRs:
- https://github.com/rancher/image-mirror/pull/471
- https://github.com/rancher/istio-installer/pull/66

Made a fix for kiali version on image-mirror:
- https://github.com/rancher/image-mirror/pull/476

Related Issue:
- https://github.com/rancher/rancher/issues/42905

### CODE REVIEW

File: packages/rancher-istio/1.19/rancher-kiali-server/generated-changes/patch/templates/rolebinding.yaml.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
+--- charts-original/templates/rolebinding.yaml
```
Comment (diogoasouza) [inline packages/rancher-istio/1.19/rancher-kiali-server/generated-changes/patch/templates/rolebinding.yaml.patch]: What is the purpose of this change here? We don't have it in 1.18

File: packages/rancher-istio/1.19/rancher-kiali-server/generated-changes/patch/templates/oauth.yaml.patch [hunk 2]
```diff
@@ -0,0 +1,12 @@
+--- charts-original/templates/oauth.yaml
```
Comment (diogoasouza) [inline packages/rancher-istio/1.19/rancher-kiali-server/generated-changes/patch/templates/oauth.yaml.patch]: Same here

File: packages/rancher-istio/1.19/rancher-kiali-server/generated-changes/patch/values.yaml.patch [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
++
+ #
+ # Settings that mimic the Kiali CR which are placed in the ConfigMap.
+ # Note that only those values used by the Helm Chart will be here.
+@@ -29,7 +32,7 @@
+   # This only limits what Kiali will attempt to see, but Kiali Service Account has permissions to see everything.
+   # For more control over what the Kial Service Account can see, use the Kiali Operator
+   accessible_namespaces:
+-  - "**"
++    - "**"
```
Comment (diogoasouza) [inline packages/rancher-istio/1.19/rancher-kiali-server/generated-changes/patch/values.yaml.patch]: This change as well

File: packages/rancher-istio/1.19/rancher-kiali-server/generated-changes/patch/values.yaml.patch [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+   #private_key_file:
+ 
+@@ -91,8 +95,8 @@
+   certificates_information_indicators:
+     enabled: true
+     secrets:
+-    - cacerts
+-    - istio-ca-secret
++      - cacerts
++      - istio-ca-secret
```
Comment (diogoasouza) [inline packages/rancher-istio/1.19/rancher-kiali-server/generated-changes/patch/values.yaml.patch]: same

File: packages/rancher-istio/1.19/rancher-kiali-server/generated-changes/patch/values.yaml.patch [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+   clustering:
+     autodetect_secrets:
+       enabled: true
+@@ -100,7 +104,7 @@
+     clusters: []
+   disabled_features: []
+   validations:
+-    ignore: ["KIA1301"]
```
Comment (diogoasouza) [inline packages/rancher-istio/1.19/rancher-kiali-server/generated-changes/patch/values.yaml.patch]: same

File: packages/rancher-istio/1.19/rancher-istio/charts/values.yaml [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    tag: 1.19.0-distroless
+  defaultPodDisruptionBudget:
+    enabled: true
+
+# Kiali subchart from rancher-kiali-server
+kiali:
+  enabled: true
+  # If you wish to change the authentication you can check the options in the Kiali documentation https://kiali.io/docs/configuration/authentication/
+  auth:
+    strategy: anonymous
```
Comment (diogoasouza) [inline packages/rancher-istio/1.19/rancher-istio/charts/values.yaml]: Can you change this from `anonymous` to `token`?

**Review by diogoasouza — CHANGES_REQUESTED** [review]

File: packages/rancher-istio/1.19/rancher-istio/charts/values.yaml (hunk 6 above)
Comment (nicholasSUSE) [reply packages/rancher-istio/1.19/rancher-istio/charts/values.yaml]: I just changed it, could you explain to me why are we changing this?

**Review by rohitsakala — APPROVED** [review]

**Comment by rohitsakala** [comment]
@nicholasSUSE @diogoasouza The strategy token change is a major thing and users can be easily confused. I would say 

- Add a UI ticket to show a warning message that the default strategy for kiali is token and they can change it if they want to through helm values.yaml or point them to the token strategy docs of kiali. 
- @nicholasSUSE we also need to test if step 1 is possible as in check if the user is able to change from token to anonymous while installing the chart or upgrading the chart.

**Comment by nicholasSUSE** [comment]
> @nicholasSUSE @diogoasouza The strategy token change is a major thing and users can be easily confused. I would say
> 
> * Add a UI ticket to show a warning message that the default strategy for kiali is token and they can change it if they want to through helm values.yaml or point them to the token strategy docs of kiali.
> * @nicholasSUSE we also need to test if step 1 is possible as in check if the user is able to change from token to anonymous while installing the chart or upgrading the chart.

ok I will test it

**Review by diogoasouza — APPROVED** [review]

**Comment by nicholasSUSE** [comment]
> @nicholasSUSE @diogoasouza The strategy token change is a major thing and users can be easily confused. I would say
> 
> * Add a UI ticket to show a warning message that the default strategy for kiali is token and they can change it if they want to through helm values.yaml or point them to the token strategy docs of kiali.
> * @nicholasSUSE we also need to test if step 1 is possible as in check if the user is able to change from token to anonymous while installing the chart or upgrading the chart.

Both Auth strategies are working now
